propagation/trace_context.go

```diff
@@ -59,12 +60,27 @@ func (tc TraceContext) Inject(ctx context.Context, carrier TextMapCarrier) {
 	// Clear all flags other than the trace-context supported sampling bit.
 	flags := sc.TraceFlags() & trace.FlagsSampled
 
-	h := fmt.Sprintf("%.2x-%s-%s-%s",
-		supportedVersion,
-		sc.TraceID(),
-		sc.SpanID(),
-		flags)
-	carrier.Set(traceparentHeader, h)
+	var sb strings.Builder
+	sb.Grow(2 + 32 + 16 + 2 + 3)
+	_, _ = sb.WriteString(versionPart)
+	traceID := sc.TraceID()
+	spanID := sc.SpanID()
+	flagByte := [1]byte{byte(flags)}
+	writeTraceParent(&sb, traceID[:], spanID[:], flagByte[:])
+	carrier.Set(traceparentHeader, sb.String())
+}
+
+func writeTraceParent(sb *strings.Builder, srcs ...[]byte) {
+	for _, src := range srcs {
+		_, _ = sb.WriteByte(parentDelimiter[0])
```

Comment:
```suggestion
		_ = sb.WriteByte(parentDelimiter[0])
```
strings.builder.WriteByte func has only one return value, please fix it.

Reply:
thanks